### PR TITLE
[Audio] Fix issues in Tinyalsa and AudioUTC for MMAP

### PR DESCRIPTION
**Pull request by kishore-sn**
- Modified logic in mmap begin and commit api's
- Fixed assert and error checking procedures in audio UTC for mmap api's
- Fixed flag and structure variables in tinyalsa

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/tinyalsa/tinyalsa.c:1453: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/tinyalsa/tinyalsa.c:1453: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
Could you write what is the issue and what is the reason in commit description? Empty description is not acceptable to merge.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/tinyalsa/tinyalsa.c [hunk 1]
```diff
@@ -129,11 +129,11 @@ struct pcm {
 	/** Flags that were passed to @ref pcm_open */
 	unsigned int flags;
 	/** Whether the PCM is running or not */
-	int running:1;
+	unsigned char running;
```
Comment (Taejun-Kwon) [inline framework/src/tinyalsa/tinyalsa.c]: I think we don't have to change this

File: apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c [hunk 2]
```diff
@@ -1381,7 +1395,7 @@ static void utc_audio_pcm_mmap_write_p(void)
 		num_read = read(fd, buffer, size);
 		if (num_read > 0) {
 			ret = pcm_mmap_write(g_pcm, buffer, pcm_bytes_to_frames(g_pcm, num_read));
-			TC_ASSERT_GEQ_CLEANUP("pcm_mmap_write", ret, 0, clean_all_data(fd, buffer));
+			TC_ASSERT_CLEANUP("pcm_mmap_write", ret >= 0 || ret == -EPIPE, clean_all_data(fd, buffer));
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: pcm_mmap_write also byte based operation

Comment (Taejun-Kwon) [reply apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: Also below should be added.
if (ret == -EPIPE) {
    pcm_prepare(); //add this later
    continue; //for now
}

File: framework/src/tinyalsa/tinyalsa.c [hunk 3]
```diff
@@ -1526,9 +1520,7 @@ int pcm_mmap_transfer(struct pcm *pcm, const void *buffer, unsigned int bytes)
 		if ((unsigned int)avail == 0) {
 			err = pcm_wait(pcm, -1);
 			if (err < 0) {
-				pcm->prepared = 0;
-				pcm->running = 0;
-				return oops(pcm, errno, "wait error: avail 0x%x\n", avail);
+				return err;
```
Comment (Taejun-Kwon) [inline framework/src/tinyalsa/tinyalsa.c]: current code is right.
Also we jsut return xrun msg without change value of pcm->prepared and pcm->running in pcm_writei, pcm_readi.
Could you check this once again?

File: framework/src/tinyalsa/tinyalsa.c (hunk 1 above)
Comment (kishore-sn) [reply framework/src/tinyalsa/tinyalsa.c]: ok

File: apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c (hunk 2 above)
Comment (kishore-sn) [reply apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: I fixed the code for your first comment.
Regarding the second comment, the loop will anyway continue as of now. I will add the pcm_prepare check later.

File: framework/src/tinyalsa/tinyalsa.c (hunk 3 above)
Comment (kishore-sn) [reply framework/src/tinyalsa/tinyalsa.c]: I will change this after we decide on how to handle the xrun. I think the pcm will continue to run even after an xrun.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**